src/main/java/commands/SearchCommand.java

```diff
@@ -48,13 +67,31 @@ public void setUserAction(String userAction) throws InvalidCommandException {
         }
     }
 
-    private void printSearchHeading() {
-        if (isFirstMatch()) {
-            System.out.println("The exercise(s) containing keywords"
-                    + ui.getColorText(TextColor.COLOR_YELLOW, " [" + userArguments + "] ")
-                    + "is(are) listed below.");
-            ui.printLine();
+    private void printSearchHeading() throws InvalidCommandException {
+        switch (userAction) {
+        case SEARCH_EXERCISE_ACTION_KEYWORD:
+            if (isFirstMatch()) {
+                printHeadingMessage("exercise");
+            }
+            break;
+        case SEARCH_PLAN_ACTION_KEYWORD:
+            if (isFirstMatch()) {
+                printHeadingMessage("plan");
+            }
+            break;
+        default:
+            logger.log(Level.WARNING, "User has entered an invalid search exercise command action.");
+            String className = this.getClass().getSimpleName();
+            throw new InvalidCommandException(className, InvalidCommandException.INVALID_ACTION_ERROR_MSG);
         }
+
```

Comment:
Perhaps this extra empty line can be removed?

Reply:
Yes, I didn't see that, thank you!